### DISCUSSION TITLE
Is there any good example on how to get the data with fetch from remote server and updating the store

### DISCUSSION THREAD

**Discussion by durga73** [ai/nanostores#150] [General]
Looking for an excellent example of integrating AJAX/Fetch calls within the store. Adding any example of how to get the data and how to set the data to the server with all the store events (like onMount). 
Thanks,
Durga

**Comment by ai** [comment]
Something like this:

```ts
import { atom, onMount } from 'nanostores'

type Posts = { isLoading: true } | { isLoading: false, posts: Post[] }

let posts = atom<Posts>({ isLoading: true })

onMount(posts, () => {
  fetch('/posts').then(async response => {
    let data = await response.json()
    posts.set({ isLoading: false, posts: data })
  })
  return () => {
    posts.set({ isLoading: true })
  }
})
```

**Reply by durga73** [reply]
Thanks for the example, but how do the React components render once we fetch the data ?? Does the onMount gets called on component mount ??

**Reply by ai** [reply]
> Does the onMount gets called on component mount

Yes

> how do the React components render once we fetch the data

On first render store will return `{ isLoading: true }` and your will render loader. When `fetch()` will be finished, store will be updated with the data and React component will be re-rendered.

```js
export const Posts = () => {
  let list = useStore(posts)
  if (list.isLoading) {
    return <Loader>
  }
  return <ul>{list.posts.map(post => <li>{post.title}</li>)}</ul>
}
```
